### PR TITLE
fix: update test packages to fix tests on macos-latest

### LINKED ISSUE #1372: Test packages need update to please the new macos-latest.
**Describe the bug**

https://github.com/pypa/pipx/actions/runs/8834099473/job/24254995156

**How to reproduce**

**Expected behavior**

### PR DESCRIPTION
**Pull request by huxuan**
- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes

Fix #1372 

1. Update some packages to newer version to install on macos-latest successfully.
2. Update tests package lists
4. Update pylint related tests due to metadata change

## Test plan

Tested by running

```
# command(s) to exercise these changes
nox -s tests
```

### CODE REVIEW

**Comment by huxuan** [comment]
It seems that `jupyter` and `requests` are too old to be installed on macos-latest

Any ideas to deal with it?

**Comment by huxuan** [comment]
emmm.... Seems updating some packages will not fix the problem, they introduce more failures. 

any ideas?

**Comment by dukecat0** [comment]
Just fixed it by deleting the caches. 👍

File: changelog.d/1372.bugfix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Update test packages to fix tests on macos-latest
```
Comment (dukecat0) [inline changelog.d/1372.bugfix.md:1]: This is not necessary as it has no effects on end users.

Comment (dukecat0) [inline changelog.d/1372.bugfix.md:1]:
```suggestion
```

**Comment by huxuan** [comment]
> Just fixed it by deleting the caches. 👍

Hi @dukecat0, thanks for the help!

Could you explain a little bit why deleting the caches will fix the tests? Actually, I spend almost the whole afternoon trying to find out why the tests keep failing. :-(

**Comment by huxuan** [comment]
@dukecat0 Thanks so much for the detailed explanation!

Actually, I do not have a macos so I thought it might be some kind of compatibility issue. You save my day!

**Review by dukecat0 — APPROVED** [review]
Thanks!

**Pull request merged**